Xamarin.Forms.Platform.UAP/SwitchRenderer.cs

```diff
@@ -53,5 +69,54 @@ void UpdateFlowDirection()
 		{
 			Control.UpdateFlowDirection(Element);
 		}
+
+		void UpdateOnColor()
+		{
+			if (Control == null)
+				return;
+
+			var grid = Control.GetChildren<Windows.UI.Xaml.Controls.Grid>().FirstOrDefault();
+			var groups = Windows.UI.Xaml.VisualStateManager.GetVisualStateGroups(grid);
+			foreach (var group in groups)
+			{
+				if (group.Name != "CommonStates")
+					continue;
+
+				foreach (var state in group.States)
+				{
+					if (state.Name != "PointerOver")
+						continue;
+
+					foreach (var timeline in state.Storyboard.Children.OfType<ObjectAnimationUsingKeyFrames>())
+					{
+						var property = Storyboard.GetTargetProperty(timeline);
+						var target = Storyboard.GetTargetName(timeline);
+						if ((target == "SwitchKnobBounds") && (property == "Fill"))
+						{
+							var frame = timeline.KeyFrames.First();
+
+							if (_originalOnColor == null)
+								_originalOnColor = frame.Value;
+
+							if (Element.IsSet(Switch.OnColorProperty) && !Element.OnColor.IsDefault)
+								frame.Value = new SolidColorBrush(Element.OnColor.ToWindowsColor()) { Opacity = .7 };
+							else
+								frame.Value = _originalOnColor;
+							break;
+						}
+					}
+				}
+			}
+
+			var rect = Control.GetDescendantsByName<Windows.UI.Xaml.Shapes.Rectangle>("SwitchKnobBounds").FirstOrDefault();
+
+			if (_originalOnColorBrush == null)
+				_originalOnColorBrush = rect.Fill;
```

Comment:
if `rect` cannot be null, then you should call `First` instead `FirstOrDefault` in LINQ

Reply:
Done